src/test/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepRjrTest.java

```diff
@@ -0,0 +1,108 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.workflow.support.steps;
+
+import hudson.model.Executor;
+import hudson.model.Queue;
+import hudson.model.Slave;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.support.steps.input.InputAction;
+import org.jenkinsci.plugins.workflow.support.steps.input.InputStepExecution;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.InboundAgentRule;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.RealJenkinsRule;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.emptyArray;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.notNullValue;
+
+public class ExecutorStepRjrTest {
+
+    @Rule
+    public RealJenkinsRule rjr = new RealJenkinsRule()
+            .withLogger(ExecutorStepExecution.class, Level.FINE)
+            .withLogger(ExecutorStepDynamicContext.class, Level.FINE)
+            .javaOptions("-Dorg.jenkinsci.plugins.workflow.support.pickles.ExecutorPickle.timeoutForNodeMillis=" + TimeUnit.SECONDS.toMillis(30));
+
+    @Rule
+    public InboundAgentRule iar = new InboundAgentRule();
+
+    @Test
+    public void restartDuringStepResumption() throws Throwable {
+        try {
+            rjr.startJenkins();
+            iar.createAgent(rjr, "custom-label");
+            rjr.runRemotely(ExecutorStepRjrTest::restartDuringStepResumption_start);
+            rjr.stopJenkins();
+            rjr.then(ExecutorStepRjrTest::restartDuringStepResumption_firstRestart);
+            rjr.then(ExecutorStepRjrTest::restartDuringStepResumption_secondRestart);
+        } finally {
+            iar.stop("custom-label");
+        }
+    }
+
+    private static void restartDuringStepResumption_start(JenkinsRule r) throws Exception {
+        WorkflowJob p = r.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition("node('custom-label') { echo 'node step started'; input('test') }", true));
+        WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+        await().atMost(15, TimeUnit.SECONDS).until(() -> b.getAction(InputAction.class), notNullValue());
```

Comment:
FWIW `RestartableJenkinsRule` + `SemaphoreStep` would likely be easier to work with, at least now that you have fixed `static runningTasks`.

Reply:
I think there was some other problem that lead me to use `RealJenkinsRule` but I do not remember exactly. I can check again.

Reply:
Seems to work fine, although the node takes long enough to reconnect that `TIMEOUT_FOR_NODE_MILLIS` needs to be increased (not sure exactly why). See [10215a6](https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/344/commits/10215a67c84fa102ad5d39481a72ab5a06a7d4b7).